File: oracle/fsharp/create/oracle_create.fs
// -------------------------------------------------------------------
//
//	oracle_create.fs
//
//					May/23/2011
//
// -------------------------------------------------------------------
open System
open System.Data
open System.Data.OracleClient

// -------------------------------------------------------------------
let table_drop_proc (conn: OracleConnection) =
  printfn "*** table_drop_proc ***"
  let sql_str = "drop table cities"
  printfn "%s" sql_str
  let cmd = new OracleCommand (sql_str, conn )
  let rowsAffected = cmd.ExecuteNonQuery()
  printfn "rowsAffected = %d" rowsAffected
  printfn "*** table_drop_proc *** finish";;

// -------------------------------------------------------------------
let table_create_proc (conn: OracleConnection) =
  printfn "*** table_create_proc ***"
  let sql_str = "create TABLE cities (" + "id varchar(10) NOT NULL PRIMARY KEY," + "name varchar(20)," + "population int," + "date_mod date)"
  printfn "%s" sql_str
  let cmd = new OracleCommand (sql_str, conn )
  let rowsAffected = cmd.ExecuteNonQuery()
  ();;

// -------------------------------------------------------------------
let data_row_proc (conn: OracleConnection,id_a,name,population_a,str_date) =
  let fmt_aa = "insert into cities (id, Name, Population, date_mod) values ('"
  let fmt_bb = id_a + "','" + name + "'," + population_a.ToString ()
  let fmt_cc = ",'" + str_date + "')"
  let sql_str = fmt_aa + fmt_bb + fmt_cc
  let cmd = new OracleCommand (sql_str, conn )
  let rowsAffected = cmd.ExecuteNonQuery()
  ();;

// -------------------------------------------------------------------
let data_insert_proc (conn: OracleConnection) =
  printfn "*** data_insert_proc *** start ***"

  data_row_proc (conn,"t0131","函館",15837,"1997-6-21")
  data_row_proc (conn,"t0132","札幌",27968,"1997-5-19")
  data_row_proc (conn,"t0133","帯広",44624,"1997-2-7")
  data_row_proc (conn,"t0134","釧路",79186,"1997-7-19")
  data_row_proc (conn,"t0135","旭川",64512,"1997-11-2")
  data_row_proc (conn,"t0136","北見",25275,"1997-10-7")
  data_row_proc (conn,"t0137","室蘭",19293,"1997-9-9")
  data_row_proc (conn,"t0138","根室",42857,"1997-5-22")
  data_row_proc (conn,"t0139","稚内",41321,"1997-7-18")

  ();;

// -------------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =
  printfn "*** 開始 ***"

  let str_connect = "User Id=scott;Password=tiger;Data Source=spn109:1521/xe"

  let conn = new OracleConnection (str_connect)

  conn.Open ()

  table_drop_proc (conn)

  table_create_proc (conn)

  data_insert_proc (conn)

  conn.Close ()

  printfn "*** 終了 ***"

  0

// -------------------------------------------------------------------

File: oracle/csharp/create/oracle_create.cs
// -------------------------------------------------------------------
/*
	oracle_create.cs

					Aug/17/2015


*/
// -------------------------------------------------------------------
using	System;
using	System.Data;
using	System.Data.OracleClient;
using	System.Text;

// -------------------------------------------------------------------
class oracle_create
{
// -------------------------------------------------------------------
static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	string str_connect =
		"User Id=scott;Password=tiger;Data Source=oracle_home:1521/xe";

	OracleConnection connection = new OracleConnection (str_connect);

	Console.WriteLine ("*** pppp ***");
	connection.Open ();
	Console.WriteLine ("*** qqqq ***");
 
	oracle_manipulate.table_drop_proc (connection);
	Console.WriteLine ("*** ffff ***");
	oracle_manipulate.table_create_proc (connection);
	Console.WriteLine ("*** gggg ***");

	DataTable dtable = oracle_manipulate.oracle_data_fetch_proc
		(connection);

	Console.WriteLine ();

	data_prepare_proc (connection);

	connection.Close ();

	table_manipulate.display_proc (dtable);

	Console.WriteLine ("*** 終了 ***");
}

// -------------------------------------------------------------------
static void data_prepare_proc (OracleConnection connection)
{
	insert_record_proc (connection,"t0131","函館",41537,"2009-3-18");
	insert_record_proc (connection,"t0132","札幌",37451,"2009-6-14");
	insert_record_proc (connection,"t0133","帯広",52542,"2009-5-7");
	insert_record_proc (connection,"t0134","釧路",33391,"2009-9-12");
	insert_record_proc (connection,"t0135","旭川",76198,"2009-8-19");
	insert_record_proc (connection,"t0136","北見",28795,"2009-1-21");
	insert_record_proc (connection,"t0137","室蘭",99854,"2009-2-24");
	insert_record_proc (connection,"t0138","根室",15476,"2009-5-22");
	insert_record_proc (connection,"t0139","稚内",67649,"2009-10-17");
}

// -------------------------------------------------------------------
static void	insert_record_proc
	(OracleConnection connection,
	String id,String name,int population,String date_mod)
{
	StringBuilder sb_sql = new StringBuilder
	("insert into cities (id, name, population, date_mod) values ('"
		+ id + "','"
		+ name + "',"
		+ population + ",'"
		+ date_mod + "')");

	String str_sql = sb_sql.ToString ();

	OracleCommand command = new OracleCommand (str_sql,connection);
	int rowsAffected = command.ExecuteNonQuery ();
	Console.WriteLine ("rowsAffected = " + rowsAffected);

}

// -------------------------------------------------------------------
}

// -------------------------------------------------------------------
